org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLCompletionSettings.java

```diff
@@ -26,13 +26,16 @@
 
 	private boolean autoCloseRemovesContent;
 
-	public XMLCompletionSettings(boolean autoCloseTags, boolean autoCloseRemovesContent) {
+	private boolean fileCompletion;

```

Comment:
This settings belongs to the file path extension, so please don't add that in this class. I suggest that you see sample like https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/ContentModelSettings.java

In other words, you should create a FilePathSettings in https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/general/settings

Please rename general to filepath.